crates/rpc/rpc-types/src/eth/filter.rs

```diff
@@ -909,6 +911,52 @@ impl From<jsonrpsee_types::SubscriptionId<'_>> for FilterId {
         }
     }
 }
+/// Specifies the kind of information you wish to receive from the `eth_newPendingTransactionFilter`
+/// RPC endpoint.
+///
+/// When this type is used in a request, it determines whether the client wishes to receive:
+/// - Only the transaction hashes (`Hashes` variant), or
+/// - Full transaction details (`Full` variant).
+#[derive(Debug)]
+pub enum PendingTransactionFilterKind {
+    /// Receive only the hashes of the transactions.
+    Hashes,
+
+    /// Receive full details of the transactions.
+    Full,
+}
+
+impl Serialize for PendingTransactionFilterKind {
+    /// Serializes the `PendingTransactionFilterKind` into a boolean value:
+    /// - `false` for `Hashes`
+    /// - `true` for `Full`
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match self {
+            PendingTransactionFilterKind::Hashes => false.serialize(serializer),
+            PendingTransactionFilterKind::Full => true.serialize(serializer),
+        }
+    }
+}
+
+impl<'a> Deserialize<'a> for PendingTransactionFilterKind {
+    /// Deserializes a boolean value into `PendingTransactionFilterKind`:
+    /// - `false` becomes `Hashes`
+    /// - `true` becomes `Full`
+    fn deserialize<D>(deserializer: D) -> Result<PendingTransactionFilterKind, D::Error>
+    where
+        D: Deserializer<'a>,
+    {
+        let val = bool::deserialize(deserializer)?;
+        if val {
+            Ok(PendingTransactionFilterKind::Full)
+        } else {
+            Ok(PendingTransactionFilterKind::Hashes)
+        }
```

Comment:
here we also need to handle the None case so this should be `Option::<bool>::deserialize` where `None` means hashes